node-notes-cli/app.ts

```diff
@@ -0,0 +1,75 @@
+import { writeFile, readFile } from 'node:fs/promises';
+
+const operation: string = process.argv[2];
+
+type Data = {
+  nextId: number;
+  notes: Record<string, string>;
+};
+
+const obj: Data = JSON.parse(await readFile('data.json', 'utf8'));
+async function updateJSON(): Promise<void> {
+  await writeFile('data.json', JSON.stringify(obj));
+}
+
+switch (operation) {
+  case 'read':
+    read();
+    break;
+  case 'create':
+    create();
+    break;
+  case 'update':
+    update();
+    break;
+  case 'delete':
+    remove();
+    break;
+  default:
+    throw Error('Invalid Operation');
+}
+
+async function read(): Promise<void> {
+  try {
```

Comment:
You could clean this up by removing try/catch from these functions and putting it around your switch. (You'd need to await all these functions in the case.)